src/gui/raster/qgsrasterminmaxwidget.h

```diff
@@ -32,6 +32,11 @@ class GUI_EXPORT QgsRasterMinMaxWidget: public QWidget, private Ui::QgsRasterMin
     void setExtent( const QgsRectangle & theExtent ) { mExtent = theExtent; }
 
     void setBands( const QList<int> & theBands ) { mBands = theBands; }
+    /** Return the extent selected by the user.
+        Either an empty extent for 'full' or the current visible extent. */
+    QgsRectangle extent() { QgsRectangle myExtent; return mCurrentExtentRadioButton->isChecked() ? mExtent : myExtent; }
```

Comment:
Could be simplified to just

````
QgsRectangle extent() { return mCurrentExtentRadioButton->isChecked() ? mExtent : QgsRectangle(); }
````

This would avoid an unnecessary QgsRectangle construction.